statement.go

```diff
@@ -33,28 +34,56 @@ func (stmt *snowflakeStmt) NumInput() int {
 func (stmt *snowflakeStmt) ExecContext(ctx context.Context, args []driver.NamedValue) (driver.Result, error) {
 	logger.WithContext(stmt.sc.ctx).Infoln("Stmt.ExecContext")
 	result, err := stmt.sc.ExecContext(ctx, stmt.query, args)
-	stmt.lastQueryID = result.(SnowflakeResult).GetQueryID()
+	if err != nil {
+		return nil, err
+	}
+	r, ok := result.(SnowflakeResult)
```

Comment:
Ok, so I definitely agree on lines 37-39, but did you have a case in which this cast didn't succeed?

Reply:
Not really. However, to improve the codebase's resilience, it would be beneficial to address potential interface conversion panics proactively. There are documented instances of such issues occurring in the project:
https://github.com/snowflakedb/gosnowflake/issues/879
https://github.com/snowflakedb/gosnowflake/issues/940

Implementing type checks could preemptively ensure stability and enable graceful error handling, thereby preventing similar issues from surfacing.